WordPress/gutenberg
# UnitControl: Use `hideHTMLArrows` prop to hide spin buttons

## What?

Uses NumberControl's `hideHTMLArrows` prop to hide the spin buttons, instead of custom CSS.

## Why?

While going through the NumberControl types (#43791), I noticed that UnitControl was adding its own CSS styles to hide the spin buttons, when in fact NumberControl provided that option via the `hideHTMLArrows` prop.

## How?

Remove the CSS overrides and use the designated prop.

## Testing Instructions

1. `npm run storybook:dev` and see the UnitControl story.
2. Spin buttons should be hidden, even when `disableUnit` is true. Check Chrome/Firefox/Safari.
